tss-esapi/tests/integration_tests/error_tests/return_code_tests/tpm_tests/tpm_format_zero_tests/tpm_format_zero_warning_tests.rs

```diff
@@ -230,3 +230,44 @@ fn test_display_implementation() {
         TpmFormatZeroWarning::NvUnavailable
     );
 }
+
+macro_rules! test_error_number_method {
+    (TpmFormatZeroWarning::$zero_warning:ident) => {
+        let zero_warning_rc = TpmFormatZeroWarningResponseCode::from(TpmFormatZeroWarning::$zero_warning);
+        assert_eq!(
+            TpmFormatZeroWarning::$zero_warning,
+            zero_warning_rc.error_number(),
+            "The TpmFormatZeroWarningResponseCode `error_number()` method did not return the expected TpmFormatZeroWarning"
+        );
+    };
+}
+
+#[test]
+fn test_error_number() {
+    test_error_number_method!(TpmFormatZeroWarning::ContextGap);
+    test_error_number_method!(TpmFormatZeroWarning::ObjectMemory);
+    test_error_number_method!(TpmFormatZeroWarning::SessionMemory);
+    test_error_number_method!(TpmFormatZeroWarning::ObjectHandles);
+    test_error_number_method!(TpmFormatZeroWarning::Locality);
+    test_error_number_method!(TpmFormatZeroWarning::Yielded);
+    test_error_number_method!(TpmFormatZeroWarning::Canceled);
+    test_error_number_method!(TpmFormatZeroWarning::Testing);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH0);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH1);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH2);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH3);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH4);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH5);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceH6);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS0);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS1);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS2);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS3);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS4);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS5);
+    test_error_number_method!(TpmFormatZeroWarning::ReferenceS6);
+    test_error_number_method!(TpmFormatZeroWarning::NvRate);
+    test_error_number_method!(TpmFormatZeroWarning::Lockout);
+    test_error_number_method!(TpmFormatZeroWarning::Retry);
+    test_error_number_method!(TpmFormatZeroWarning::NvUnavailable);
```

Comment:
:thinking: Hmm... too bad there's no easy way to [enumerate all enum variants](https://stackoverflow.com/q/21371534). This would take a lot of this boilerplate code out (I just had to open the source code to see if all variants were being covered).

Reply:
I'd add it only during testing (with `#[cfg_attr(test, ...)]`) and make strum a dev dependency... not sure about the full solution though so I don't want to push you in that direction.

Reply:
I t would be interesting and certainly something I would like to look into in another PR.

Reply:
It is unfortunately not that easy. Because the integration tests tests everything as if the thing tested are an actual crate. So in order to to do this we would probably have to add a ```integration-tests``` feature and then ccompile it using ```--features integration-tests``` when run it in ci.